src/parsers/ddi_parser.jl

```diff
@@ -0,0 +1,189 @@
+
+@kwdef mutable struct DDIVariable
+    name::String = ""
+    position_start::Int64 = 9999
+    position_end::Int64 = 9999
+    position_width::Int64 = 9999 
+    labl::String = ""
+    txt::String = ""
+    dcml::Int64 = 9999
+    vartype::String = ""
+end
+
+@kwdef mutable struct DDIInfo
+    filepath::String
+    conditions::String = ""
+    citation::String = ""
+    ipums_project::String = ""
+    extract_notes::String = ""
+    extract_date::String = ""
+    variable_info::Vector{DDIVariable} = DDIVariable[]
+    _xml_doc::EzXML.Document = EzXML.XMLDocument()
+    _ns::String = ""
+end
+
+
+
+#=
+
+Set of XPaths for variables in DDI (.xml) files
+
+=#
+
+EXTRACT_CONDITIONS = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:conditions"
+EXTRACT_CITATION = "/x:codeBook/x:stdyDscr/x:dataAccs/x:useStmt/x:citReq"
+EXTRACT_IPUMS_PROJECT = "/x:codeBook/x:stdyDscr/x:citation/x:serStmt/x:serName"
+EXTRACT_NOTES = "/x:codeBook/x:stdyDscr/x:notes"
+EXTRACT_DATE = "/x:codeBook/x:stdyDscr/x:citation/x:prodStmt/x:prodDate/@date"
+
+VAR_NAME_XPATH = "/x:codeBook/x:dataDscr/x:var/@name"
+VAR_STARTPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@StartPos"
+VAR_ENDPOS_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@EndPos"
+VAR_WIDTH_XPATH = "/x:codeBook/x:dataDscr/x:var/x:location/@width"
+VAR_LABL_XPATH = "/x:codeBook/x:dataDscr/x:var/x:labl"
+VAR_TXT_XPATH = "/x:codeBook/x:dataDscr/x:var/x:txt"
+VAR_DCML_XPATH = "/x:codeBook/x:dataDscr/x:var/@dcml"
+VAR_TYPE_XPATH = "/x:codeBook/x:dataDscr/x:var/x:varFormat/@type"
+VAR_INTERVAL_XPATH = "/x:codeBook/x:dataDscr/x:var/@intrvl"
+VAR_CATEGORY_XPATH = "/x:codeBook/x:dataDscr/x:var/x:catgry"
```

Comment:
This looks quite intense! Could you both:

- Move this to the `constants.jl` file? 
- Give a brief description about what each of these things are and where they came from?

Thanks!